nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/util/FlowDifferenceFilters.java

```diff
@@ -58,6 +71,12 @@ public class FlowDifferenceFilters {
      * @return <code>true</code> if the change is an environment-specific change, <code>false</code> otherwise
      */
     public static boolean isEnvironmentalChange(final FlowDifference difference, final VersionedProcessGroup localGroup, final FlowManager flowManager) {
+        return isEnvironmentalChange(difference, localGroup, flowManager, EnvironmentalChangeContext.empty());
+    }
+
+    public static boolean isEnvironmentalChange(final FlowDifference difference, final VersionedProcessGroup localGroup, final FlowManager flowManager,
+                                                final EnvironmentalChangeContext context) {
+        final EnvironmentalChangeContext evaluatedContext = context == null ? EnvironmentalChangeContext.empty() : context;
```

Comment:
I recommend requiring the `EnvironmentalChangeContext` for this method, instead of substituting an `empty()` context.